CMakeLists.txt

```diff
@@ -2,25 +2,116 @@ cmake_minimum_required (VERSION 3.2.0 FATAL_ERROR)
 
 project (taocpp-postgres VERSION 1.0.0 LANGUAGES CXX)
 
-set (CMAKE_CXX_STANDARD 11)
-set (CMAKE_CXX_STANDARD_REQUIRED ON)
+if (EXISTS "${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
+    include("${CMAKE_BINARY_DIR}/conanbuildinfo.cmake")
+    conan_basic_setup()
+endif()
 
-# taocpp/postgres
-include ("taocpp-postgres.cmake")
-include_directories (${TAOCPP_POSTGRES_INCLUDE_DIRS})
+list (INSERT CMAKE_MODULE_PATH 0 ${CMAKE_SOURCE_DIR}/cmake)
 
-# testing
-enable_testing ()
-option (TAOCPP_POSTGRES_BUILD_TESTS "Build test programs" ON)
-if (TAOCPP_POSTGRES_BUILD_TESTS)
-  add_subdirectory (src/test/utility)
-  add_subdirectory (src/test/postgres)
-endif ()
+find_package (PostgreSQL REQUIRED)
 
-# installation directories
+# taocpp postgres options
 set (TAOCPP_POSTGRES_INSTALL_INCLUDE_DIR "include" CACHE STRING "The installation include directory")
 set (TAOCPP_POSTGRES_INSTALL_DOC_DIR "share/doc/tao/postgres" CACHE STRING "The installation doc directory")
+option (TAOCPP_POSTGRES_BUILD_TESTS "Build test programs" ON)
+
+# taocpp/postgres
+set (TAOCPP_POSTGRES_INCLUDE_DIRS ${CMAKE_CURRENT_LIST_DIR}/include)
+
+file (GLOB_RECURSE TAOCPP_POSTGRES_INCLUDE_FILES ${TAOCPP_POSTGRES_INCLUDE_DIRS}/*.hpp)
```

Comment:
According CMake community, GLOB should be avoided:

https://gist.github.com/mbinna/c61dbb39bca0e4fb7d1f73b0d66a4fd1#dont-use-fileglob-in-projects